src/cli/deploy.ts

```diff
@@ -12,7 +12,88 @@ import { sanitizeName } from "./init/utils";
 import { deployImagePullSecret } from "../lib/assets/deploy";
 import { namespaceDeploymentsReady } from "../lib/deploymentChecks";
 
-export default function (program: RootCmd) {
+export interface ImagePullSecretDetails {
+  pullSecret?: string;
+  dockerServer?: string;
+  dockerUsername?: string;
+  dockerEmail?: string;
+  dockerPassword?: string;
+}
+
+export function validateImagePullSecretDetails(details: ImagePullSecretDetails): {
+  valid: boolean;
+  error?: string;
+} {
+  if (!details.pullSecret) {
+    return { valid: true };
+  }
+
+  // https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
+  if (details.pullSecret !== sanitizeName(details.pullSecret)) {
+    return {
+      valid: false,
+      error: `Invalid --pullSecret. Must be valid name as defined in RFC 1123.`,
+    };
+  }
+
+  const missing: string[] = [];
+  if (!details.dockerEmail) {
+    missing.push("--docker-email");
+  }
+  if (!details.dockerServer) {
+    missing.push("--docker-server");
+  }
+  if (!details.dockerUsername) {
+    missing.push("--docker-username");
+  }
+  if (!details.dockerPassword) {
+    missing.push("--docker-password");
+  }
```

Comment:
This addresses the complexity requirements, so it's good to go. Although, I wonder if there's a way we could rephrase this block as "IF X" instead of "IF NOT X." Not a big deal, and shouldn't hold up the PR.

Reply:
Yeah, I don't usually prefer to not use non-negations too (teehee) but I didn't have anything new to say about those values since this was a refactor and checking anything other than existence -- a.k.a. the "don't be undefined / null / empty string" of the pre-existing behavior -- felt like it'd be changing externally-visible behavior. :man_shrugging: